lib/net/ldap.rb

```diff
@@ -540,6 +540,9 @@ def initialize(args = {})
     @base = args[:base] || DefaultTreebase
     @force_no_page = args[:force_no_page] || DefaultForceNoPage
     @encryption = args[:encryption] # may be nil
+    if !@encryption.nil? and !@encryption.is_a?(Hash)
```

Comment:
@jch @mynameisrufus Thank you for pointing out this condition. made my understanding about these operators clear :+1: